### PR TITLE
Delete poll

### PR DESCRIPTION
**Pull request by mindylou**
- Implement the ability for an admin to delete a poll (both while live and non-live)
- New socket routes for `delete` and `deleteLive`
- Looking for feedback on the delegate implementations!
- TODO: deleting polls from the non-current session in `PollsDateViewController` (will check tomorrow)

Screenshot:
simulator screen shot - iphone 7 - 2019-03-02 at 14 33 07

### CODE REVIEW

File: Clicker/ViewControllers/EditPollViewController.swift [hunk 1]
```diff
@@ -125,12 +157,24 @@ class EditPollViewController: UIViewController {
...
                 make.width.equalToSuperview().multipliedBy(editNameButtonWidthScaleFactor)
             }
+        } else {
+            reopenButton.snp.makeConstraints { make in
+                make.leading.equalToSuperview().offset(reopenButtonLeadingPadding)
+                make.centerY.equalToSuperview()
+                make.width.equalToSuperview().multipliedBy(reopenButtonWidthScaleFactor)
+            }
+
+//            reopenView.snp.makeConstraints { make in
```
Comment (kevinchan159) [inline Clicker/ViewControllers/EditPollViewController.swift]: is this being used?

File: Clicker/ViewControllers/EditPollViewController.swift [hunk 2]
```diff
@@ -141,13 +185,19 @@ class EditPollViewController: UIViewController {
...
         }
+
     }
     
     // MARK: ACTIONS
     
     @objc func deleteBtnPressed() {
-        let deleteVC = DeletePollViewController(delegate: self, session: session, userRole: userRole)
-        self.navigationController?.pushViewController(deleteVC, animated: true)
+        switch editType! {
```
Comment (kevinchan159) [inline Clicker/ViewControllers/EditPollViewController.swift]: can we unwrap with a guard let in the line before?

File: Clicker/ViewControllers/EditPollViewController.swift [hunk 3]
```diff
@@ -170,11 +223,28 @@ class EditPollViewController: UIViewController {
         exitButton.setImage(#imageLiteral(resourceName: "exit"), for: .normal)
         exitButton.addTarget(self, action: #selector(exitBtnPressed), for: .touchUpInside)
         self.navigationItem.rightBarButtonItem = UIBarButtonItem(customView: exitButton)
+
+        switch editType! {
```
Comment (kevinchan159) [inline Clicker/ViewControllers/EditPollViewController.swift]: same comment as above

File: Clicker/ViewControllers/EditPollViewController.swift (hunk 1 above)
Comment (mindylou) [reply Clicker/ViewControllers/EditPollViewController.swift]: not right now b/c the functionality for reopening is unclear - i can delete it for now

**Comment by mathewscullin** [comment]
Looks good to me!

**Pull request merged**